extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {getSocialConfig} from './amp-social-share-config';
+import {addParamsToUrl} from '../../../src/url.js';
+import {elementByTag} from '../../../src/dom.js';
+
+/** @const {number} */
+const DEFAULT_WIDTH = 60;
+/** @const {number} */
+const DEFAULT_HEIGHT = 44;
+
+/** @const {string} */
+const CONTAINER_TAG = 'span';
+/** @const {string} */
+const LINK_TAG = 'a';
+/** @const {string} */
+const JSON_TAG = 'script';
+
+class AmpSocialShare extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+
+    /** @private @const {Object} */
+    this.config_ = this.getElementConfig_(this.element);
+    AMP.assert(this.config_, 'The <script> config is required for ' +
+      '<amp-social-share> %s');
+
+    /** @private @const {!Element} */
+    this.type_ = AMP.assert(this.element.getAttribute('type'),
+        'The type attribute is required for <amp-social-share> %s',
+        this.element);
+    /** @private @const {!Object} */
+    this.typeConfig_ = getSocialConfig(this.type_);
+
+    /** @private @const {Element} */
+    this.container_ = this.renderShare_(this.config_);
```

Comment:
Why pass `this.config_` here? It seems a bit confusing that the method itself uses `this.typeConfig_` as a member property and `this.config_` as an argument.